### DISCUSSION TITLE
Unable to flash ESP32C3

### DISCUSSION THREAD

**Discussion by Leytronix** [arendst/Tasmota#12455] [Q&A] [answered]
With the specials firmware I still am not able to flash the C3 ;-(

C:\Users\Tjerk\Desktop\Tasmota-specials-firmware>esptool.py --chip esp32 --port COM6 --baud 460800 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0x0000 bootloader_dio_80m.bin 0x8000 partitions.bin 0xe000 boot_app0.bin 0x10000 tasmota32c3.bin
usage: esptool write_flash [-h] [--erase-all] [--flash_freq {keep,40m,26m,20m,80m}]
                           [--flash_mode {keep,qio,qout,dio,dout}] [--flash_size FLASH_SIZE]
                           [--spi-connection SPI_CONNECTION] [--no-progress] [--verify] [--encrypt]
                           [--ignore-flash-encryption-efuse-setting] [--compress | --no-compress]
                           <address> <filename> [<address> <filename> ...]
esptool write_flash: error: argument <address> <filename>: [Errno 2] No such file or directory: 'bootloader_dio_80m.bin'

**Comment by sfromis** [comment, marked as answer]
Well, you need to have the needed extra files downloaded first, to be able to flash them.

**Reply by sfromis** [reply]
Not sure if it is the correct one, but the file name exists here, including a relation to ESP32C3:
https://github.com/espressif/arduino-esp32/tree/master/tools/sdk/esp32c3/bin

**Comment by Jason2866** [comment]
You find all needed files here https://github.com/Jason2866/Tasmota-specials/tree/firmware/firmware/tasmota32/ESP32-C3_needed_files

**Reply by brev-dev** [reply]
For those people coming across this thread today, the needed extra files can currently be found at https://github.com/arendst/Tasmota-firmware/tree/main/static/esp32-c3.

Along with @Jason2866's unofficial firmware builds from https://github.com/Jason2866/Tasmota-specials,  they worked for me on an esp32-c3f.

**Reply by Jason2866** [reply]
Since today you can Flash via [Tasmota Web Installer](https://jason2866.github.io/Tasmota-specials)
![image](https://user-images.githubusercontent.com/24528715/130594720-ee05ac20-2130-4eda-9c4c-1e07a0242bb5.png)
